### ISSUE TITLE
`mouseWheelInput.js`: Two finger scrolling on Firefox scrolls the whole page

### ISSUE THREAD

**Issue by talkdirty** [cornerstonejs/cornerstoneTools#643]
### Prerequisites

* [x] Are you running the latest version?
* [x] Are you reporting to the correct repository?
* [x] Did you perform a cursory search?

### Description

Hi!

We noticed a peculiar thing specific to Firefox related to `mouseWheelInput.js`. Two finger scrolling (on a Laptop with a touchpad) will scroll the page and images simultaneously.
We found the potential culprit of this, which is the legacy `DOMMouseScroll` event used by `mouseWheelInput.js`. More recent versions of Firefox don't need the Firefox specific `DOMMouseScroll` anymore, and accept the generic `wheel` event. We tried replacing `DOMMouseScroll` with `wheel` (i.e., setting `const mouseWheelEvents = ['wheel']`) in `mouseWheelInput.js`, and it works perfectly with this.

I'd be happy to open a PR for this if anyone can confirm the issue. [Here is the documentation](https://developer.mozilla.org/en-US/docs/Web/Events/wheel) of wheel on MDN, along with a browser compatibility table.

Thanks a lot for this awesome project!
Best Regards

### Steps to Reproduce

1. Latest Firefox 64.0b1; Laptop with a Touchpad and two finger scrolling
2. Cornerstone viewer in a scrollable context, using `mouseWheelInput.js` for scrolling through images
3. Use two-finger scrolling on the cornerstone viewer

**Expected behavior:** Page should not scroll

**Actual behavior:** Page and images viewed in cornerstone scroll simultaneously

### Codepen With Reproduction of Issue:

- [Based on scroll example](https://codepen.io/talkdirty/pen/qJxQap)
Make the output window small enough so it scrolls, then try two finger scrolling on firefox. The whole page will scroll.

**Comment by dannyrb** [comment]
@talkdirty thanks for the reproduction and detailed issue!

Your proposed fix makes sense to me. Feel free to create a PR addressing the issue and tag me as the reviewer (or `@mention` me), and I will be happy to merge the fix  🎉

**Comment by talkdirty** [comment]
Thanks for the quick response! See the referenced PR. @swederik might probably weigh into this, as he appears to be the original author of the functionality. Note that I could only test this on recent Edge, Firefox, and Chrome. I don't know how the behavior is on previous versions and IE. Also, I think the file could now be refactored a bit (i.e., it should be retested if the hack is still relevant)

**Comment by dannyrb** [comment]
Thanks for the quick PR, @talkdirty. There's plenty of code that could be touched up, don't be shy! If you want in on the most active development, you may want to take a peak at the `vNext` branch.

I'll let this sit for a day or two to give @swederik a chance to weigh in on any particular issues he may have been addressing. As far as I can tell, the event is well supported enough now that this should be safe to do: https://developer.mozilla.org/en-US/docs/Web/Events/wheel

**Comment by swederik** [comment]
Thanks for the super detailed report. I don't mind sunsetting the old mouse scroll events, but I have this memory of trying to switch to 'wheel' a while back and finding issues with the way it was implemented by different browsers / operating systems. Some input devices fire tons of events, others squish concurrent events down to a single event and just change the amount of delta. In the end all the workarounds ended up ruining the nice momentum scrolling of the OSX trackpad (as an example).

It's also tough to test if the scrolling feels normal if the stack is too small. It helps to have a demo set to scroll through. This example by @diego0020 might help: https://rawgit.com/cornerstonejs/cornerstoneTools/master/examples/stackImagePositionOffsetSynchronizer/index.html

I would say if we are going to switch to this, we should put up a demo page and make sure it works well across all OS / input device combinations.

**Comment by talkdirty** [comment]
Yes, I'm a bit worried about the cross platform behavior of this change as well, because wheel does clearly behave differently (albeit better in all scenarios we've come across). I can feel the pain @swederik must have been having implementing `mouseWheelInput.js`. All the browser specific hacks in there speak for themselves. Let me know if there is a way I can help with testing this, I'd be happy to put some time into it.

**Comment by diego0020** [comment]
I could help with the example the spirit of hacktoberfest

**Comment by dannyrb** [comment]
From the related pull request:

> @talkdirty, alright. I've merged the fix for this to `vNext`. We have the option of backporting this to `v2`, but we would lose some browser support because I dropped listening for `'mousewheel', 'DOMMouseScroll'` and instead only listen to `wheel`. I'm okay with doing that in `vNext`, because it's the next major release, and I'm okay with the breaking change there.
> 
> Whether we should do this for `v2` isn't as cut and dry, even though I really, really hope no one is using `cornerstone` on `<= IE 9` or older versions of evergreen browsers. I would be willing to backport it if @swederik gave the 👍
> 
> I believe #643 can be addressed with a non-breaking change, but I don't have the bandwidth to look into it.
> 
> 
> *tl;dr:*
> 
> - I will look at and merge any non-breaking solutions
> - If @swederik gives a 👍, I would spend the time to merge the `vNext` solution to this branch
> - I recommend waiting for `vNext` RC or bravely porting existing code to it now if this issue is plaguing you

I'll leave this issue open for ~6-months after `vNext` becomes `v3`; if we don't have an implementation by then, I'll likely close this issue and mark as `wontfix`
